### PR TITLE
Add streaming examples to gRPC perf docs

### PR DESCRIPTION
**Pull request by JamesNK**
Note that the gRPC example that is linked to hasn't been merged yet - https://github.com/grpc/grpc-dotnet/pull/1529

### CODE REVIEW

File: aspnetcore/grpc/performance.md [hunk 1]
```diff
@@ -274,10 +274,12 @@ gRPC is a message-based RPC framework, which means:
 * The entire message is loaded into memory before gRPC can send it.
 * When the message is received, the entire message is deserialized into memory.
 
-Messages with large binary payloads can allocate byte arrays on the [large object heap](/dotnet/standard/garbage-collection/large-object-heap). Large allocations impact server performance and scalability.
+Binary payloads are allocated as a byte array. For example, a 10 megabyte binary payload allocates a 10 megabyte byte array. Messages with large binary payloads can allocate byte arrays on the [large object heap](/dotnet/standard/garbage-collection/large-object-heap). Large allocations impact server performance and scalability.
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/performance.md:277]:
```suggestion
Binary payloads are allocated as a byte array. For example, a 10 MB binary payload allocates a 10 MB byte array. Messages with large binary payloads can allocate byte arrays on the [large object heap](/dotnet/standard/garbage-collection/large-object-heap). Large allocations impact server performance and scalability.
```

Comment (Rick-Anderson) [reply aspnetcore/grpc/performance.md:277]: Or ` .. a 10 megabyte (MB) ... ` on first usage then use MB on the next occurrence.  We use MB by itself extensively and have no reported problems with that.

**Review by Rick-Anderson — APPROVED** [review]

File: aspnetcore/grpc/performance.md [hunk 2]
```diff
@@ -274,10 +274,12 @@ gRPC is a message-based RPC framework, which means:
...
 
 Advice for creating high-performance applications with large binary payloads:
 
 * **Avoid** large binary payloads in gRPC messages. A byte array larger than 85,000 bytes is considered a large object. Keeping below that size avoids allocating on the large object heap.
-* **Consider** splitting large binary payloads [using streaming](xref:grpc/client#client-streaming-call). Binary data is chunked and streamed over multiple messages.
-* **Consider** not using gRPC for large binary data. HTTP endpoints can be used alongside gRPC services. An HTTP endpoint that supports sending or receiving large files using the stream body is efficient.
+* **Consider** splitting large binary payloads [using gRPC streaming](xref:grpc/client#client-streaming-call). Binary data is chunked and streamed over multiple messages. For more information on how to stream files, see an [example of gRPC streaming file upload](https://github.com/grpc/grpc-dotnet/tree/master/examples#uploader) at the grpc-dotnet repository.
```
Comment (guardrex) [inline aspnetcore/grpc/performance.md:282]: You can if you wish ... probably a good idea ... use our INCLUDE about cross-links to ref source.

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/includes/aspnetcore-repo-ref-source-links.md

... which adds the following NOTE ...

> \> \[!NOTE]
> \> Documentation links to the ASP.NET Core reference source load the repository's \`main\` branch, which represents the product unit's current development for the next release of ASP.NET Core. To select the branch for a different release, use the \*\*Switch branches or tags\*\* dropdown list to select the branch. For example, select the \`release/6.0\` branch for the ASP.NET Core 6.0 release.

Added below this list (or for the one list item) as ...

```
[!INCLUDE[](~/includes/aspnetcore-repo-ref-source-links.md)]
```

**Review by guardrex — APPROVED** [review]

File: aspnetcore/grpc/performance.md (hunk 2 above)
Comment (guardrex) [reply aspnetcore/grpc/performance.md:282]: ... and btw ... it's `master` on that one??? I thought that they were all `main` now.

Comment (guardrex) [reply aspnetcore/grpc/performance.md:282]: Huh ... ok ... well ... it is `master` on that repo ... and the versions aren't "release" versions of ASP.NET Core. This is a bit much effort for such a small thing. I now recommend just forgetting about it in this case.

**Pull request merged**